### PR TITLE
Remove git dependency

### LINKED ISSUE #2563: Don't require Git to build binaryen
Since version 90 binaryen requires Git to be built from source. Linux/BSD distributions will usually just download the release tarball and won't build binaryen from Git directly. Please consider to remove or relax the requirement to build-depend on Git. For instance you could just change CMakelists.txt to

 set(BINARYEN_VERSION_INFO "90")

before you cut a release and create the release tarball.

### PR DESCRIPTION
**Pull request by sbc100**
Only use git to set version number if .git directory is present.
This means that for release archives the VERSION string will be used
as-is.

Fixes #2563

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
-project(binaryen C CXX)
 cmake_minimum_required(VERSION 3.1.3)
+project(binaryen LANGUAGES C CXX VERSION 90)
```
Comment (kripken) [inline CMakeLists.txt]: Will we need to change this each time we update the version? Alternatively, could this have `"unknown"` instead of `90`?

Comment (sbc100) [reply CMakeLists.txt]: Yes we will need to change this when we do releases.. although if we forget it only effects people who don't use git.. and not a way that is particularly problematic.

Comment (kripken) [reply CMakeLists.txt]: I think it would be better to report "unknown" than possibly a wrong version number, myself. But I don't feel strongly.

Comment (sbc100) [reply CMakeLists.txt]: But that means that anyone who packages binaryen from a source archive alwasy gets an "unknown" version.

I'd rather write a test that the CHANGELOG matches this version..  but I think that might be overkill.

**Pull request merged**